src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonMetadataServicesConverter.cs

```diff
@@ -70,15 +70,17 @@ internal override bool OnTryWrite(Utf8JsonWriter writer, T value, JsonSerializer
             Debug.Assert(options == jsonTypeInfo.Options);
 
             if (!state.SupportContinuation &&
+                jsonTypeInfo.HasSerializeHandler &&
                 jsonTypeInfo is JsonTypeInfo<T> info &&
```

Comment:
Currently, yes. It can be `JsonTypeInfo<SomeOtherType>` because of polymorphism. I was planning on changing that as part of a fix for #72187 but then I hit this bug. I will change in a second iteration.